js/ai/src/generate.ts

```diff
@@ -346,6 +346,8 @@ async function resolveFullToolName(
     return `/tool/${name}`;
   } else if (await registry.lookupAction(`/prompt/${name}`)) {
     return `/prompt/${name}`;
+  } else if (await registry.lookupAction(`/prompt/dotprompt/${name}`)) {
+    return `/prompt/dotprompt/${name}`;
```

Comment:
Do we want dotprompt segregated in the registry? Or should it just be part of `/prompt/` namespace?

Reply:
I don't want to... but prompt file loading is async... so we need an async initialization hook that will block registry lookup, and the only thing that can do that is plugins, which require namespacing.
Can look into adding async registry providers... so that you can register an action with a promise.
Do you foresee issues with namespacing like this?

Reply:
Since the actions themselves are async couldn't we push it down a layer? Have each action have a `loadAllPromptsAndStuff()` at the top that it awaits before it does other work?

Reply:
no, because we need action metadata to register the top action...

Reply:
we'd need a promise action registration which only requires action type/name as input, everything else provided in a promise...

Reply:
Ah, okay. I can't think of a reason this is a *real* problem, so I'll leave it be.